File: calendarview/src/main/java/al/bruno/calendar/view/observer/Subject.java
package al.bruno.calendar.view.observer;

/**
 * Created by 1sd on 3/15/18.
 */

public interface Subject<T> {
    void registerObserver(Observer<T> o);
    void removeObserver(Observer<T> o);
    void notifyObserver(T t);
}

File: app/src/main/java/al/bruno/personal/expense/adapter/observer/Subject.kt
package al.bruno.personal.expense.adapter.observer

interface Subject<T> {
    fun registerObserver(o: Observer<T>)
    fun removeObserver(o: Observer<T>)
    fun notifyObserverChanged(t: T)
    fun notifyObserverRemove(t: T);
    fun notifyObserverAdd(t: T)
}